src/StackExchange.Redis/RawResult.cs

```diff
@@ -133,6 +133,8 @@ internal RedisChannel AsRedisChannel(byte[]? channelPrefix, RedisChannel.Pattern
                         return new RedisChannel(copy, mode);
                     }
                     return default;
+                case ResultType.Error:
+                    throw new InvalidCastException("Cannot convert to RedisChannel: " + ToString());
```

Comment:
I think we should reduce this down to the raw redis error; my suggestion:

``` c#
throw new RedisServerException(GetString());
```

a few differences here:

- we're emphasizing that this is an error coming from the server
- we're keeping the error #2020 
- the use of `GetString` avoids additional words

Can we use this in all the places? with that, I think we're good